### PR TITLE
IARRC Drivers and Firmware

### PR DESCRIPTION
**Pull request by ValRat**

### CODE REVIEW

File: src/drivers/launch/steering_driver.launch [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<launch>
+
+    <!-- Launch Steering Driver -->
+    <node name="steering_driver" pkg="drivers" type="steering_driver" output="screen">
+        <!-- TODO: Descriptive COmments -->
```
Comment (RobynCastro) [inline src/drivers/launch/steering_driver.launch]: Not that important but should be "Descriptive Comments".

File: src/drivers/src/SteeringDriver.cpp [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+ * @param input_min the minimum possible value of the input range
+ * @param input_max the maximum possible value of the input range
+ * @param output_min the minimum possible value of the output range
+ * @param output_max the maximum possible value of the output range
+ *
+ * @return the mapped value
+ **/
+void map(double& input, double input_min, double input_max,
+        double output_min, double output_max) {
+    if (input < input_min) input = input_min;
```
Comment (RobynCastro) [inline src/drivers/src/SteeringDriver.cpp]: could use "input = max(input, input_min);"

File: src/drivers/src/SteeringDriver.cpp [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    else if (input > input_max) input = input_max;
```
Comment (RobynCastro) [inline src/drivers/src/SteeringDriver.cpp]: similiarily could use "input = min(input, input_max);"

File: src/firmware/olaf_control/olaf_control.ino [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+ * where each `<>` is a single byte. The degree to rotate the servo controlling
+ * the front wheels is determined from `angular z`, and the speed of the car by `linear x`
+ * (the rest of the values are discarded)
+ */
+
+#include <SoftwareSerial.h>
+#include <stdlib.h>
+#include <Servo.h>
+
+// Uncomment this to enable debug message logging - messages sent over serial will be echoed back
```
Comment (RobynCastro) [inline src/firmware/olaf_control/olaf_control.ino]: Should warn people that if DEBUGGING is activated, the robot might stop working due to timing errors.

File: src/drivers/launch/steering_driver.launch (hunk 1 above)
Comment (ValRat) [reply src/drivers/launch/steering_driver.launch]: Done

File: src/drivers/src/SteeringDriver.cpp (hunk 2 above)
Comment (ValRat) [reply src/drivers/src/SteeringDriver.cpp]: im dumb keeping this as is

File: src/drivers/src/SteeringDriver.cpp (hunk 3 above)
Comment (ValRat) [reply src/drivers/src/SteeringDriver.cpp]: im dumb keeping this as is

File: src/firmware/olaf_control/olaf_control.ino (hunk 4 above)
Comment (ValRat) [reply src/firmware/olaf_control/olaf_control.ino]: done

**Comment by garethellis0** [comment]
Why wasn't this already in `iarrc`? Did I miss it when I did the migration?

**Pull request merged**